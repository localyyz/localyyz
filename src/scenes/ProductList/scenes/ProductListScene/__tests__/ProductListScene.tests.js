import React from "react";

// custom
import ShallowRenderer from "react-test-renderer/shallow";

// local
import ProductListScene from "../";
const Component = ProductListScene.wrappedComponent;
let shallowRenderer = new ShallowRenderer();

// constants
const NAME = "ProductListScene";
const PROPS = {
  store: {}
};

describe(NAME, () => {
  it("should render properly", () => {
    shallowRenderer.render(<Component {...PROPS} />);
    const c = shallowRenderer.getRenderOutput();

    expect(c).toBeDefined();
  });

  it("should match snapshot", () => {
    shallowRenderer.render(<Component {...PROPS} />);
    const c = shallowRenderer.getRenderOutput();

    expect(c).toMatchSnapshot();
  });
});
