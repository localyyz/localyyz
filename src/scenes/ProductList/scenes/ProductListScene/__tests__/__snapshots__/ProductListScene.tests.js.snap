// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductListScene should match snapshot 1`] = `
<Provider
  filterStore={
    FilterStore {
      "_gender": undefined,
      "asyncFetch": [Function],
      "brand": undefined,
      "brands": Array [],
      "color": undefined,
      "colors": Array [],
      "discountMin": undefined,
      "fetchBrands": [Function],
      "fetchColors": [Function],
      "fetchSizes": [Function],
      "filterReaction": [Function],
      "priceMax": undefined,
      "priceMin": undefined,
      "refresh": [Function],
      "reset": [Function],
      "scrollEnabled": true,
      "searchStore": Object {},
      "setBrandFilter": [Function],
      "setColorFilter": [Function],
      "setDiscountFilter": [Function],
      "setGenderFilter": [Function],
      "setPriceFilter": [Function],
      "setScrollEnabled": [Function],
      "setSizeFilter": [Function],
      "setSortBy": [Function],
      "size": undefined,
      "sizes": Array [],
      "sortBy": "created_at",
    }
  }
  productListStore={Object {}}
>
  <Component
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <ContentCoverSlider
      backAction={[Function]}
      backColor="rgb(30, 30, 30)"
      background={
        <Component
          onLayout={[Function]}
        >
          <Header
            store={Object {}}
          />
        </Component>
      }
    >
      <SectionList
        contentContainerStyle={
          Object {
            "marginTop": 60,
            "paddingBottom": 144,
          }
        }
        data={Array []}
        disableVirtualization={false}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        keyboardShouldPersistTaps="always"
        maxToRenderPerBatch={10}
        onEndReached={[Function]}
        onEndReachedThreshold={1}
        onScroll={[Function]}
        renderItem={[Function]}
        renderSectionHeader={[Function]}
        scrollEventThrottle={16}
        sections={
          Array [
            Object {
              "data": Array [
                Array [
                  Object {},
                ],
              ],
              "renderItem": [Function],
              "title": "spacer",
            },
            Object {
              "data": Array [
                Array [
                  Object {},
                ],
              ],
              "title": "content",
            },
          ]
        }
        showsHorizontalScrollIndicator={false}
        showsVerticalScrollIndicator={false}
        stickySectionHeadersEnabled={true}
        updateCellsBatchingPeriod={50}
        windowSize={21}
      />
    </ContentCoverSlider>
  </Component>
</Provider>
`;
